src/napari_graph/base_graph.py

```diff
@@ -245,6 +247,9 @@ def __init__(
                     f"`edges` (shape: {edges.shape}) must have shape E x 2."
                 )
 
+            if self._coords is None:
+                self.add_nodes(np.unique(edges))
```

Comment:
Maybe move this to the "initialize edges" part?

Reply:
Also, shouldn't this check `coords` rather than `self._coords`?

Reply:
Both works. At this point, when `self._coords is None`, `coords` is also `None`.
I'm going to move it down to "initialize edges"